server/game/core/gameSystem/PlayerTargetSystem.ts

```diff
@@ -22,6 +25,18 @@ export abstract class PlayerTargetSystem<TContext extends AbilityContext = Abili
         return this.canAffect(event.player, event.context, additionalProperties);
     }
 
+    public override resolve(target: undefined | Player | Player[], context: TContext, triggerHandlingMode?: TriggerHandlingMode) {
+        super.resolve(target, context, triggerHandlingMode);
+    }
+
+    protected override updateEvent(event: GameEvent, player: Player, context: TContext, additionalProperties?: any): void {
```

Comment:
why override these methods but just call super ?

Reply:
Oh yeah good question - we're overriding the type of the second parameter to be `Player` instead of `any` so that the child classes are forced to use that same signature. It's definitely unclear though, I'll add a comment